worker/src/ee/cloudSpendAlerts/handleCloudSpendAlertJob.ts

```diff
@@ -0,0 +1,258 @@
+            `[CLOUD SPEND ALERTS] Triggering alert ${alert.id} for org ${orgId} - spend $${currentSpendUSD.toFixed(2)} >= threshold $${thresholdUSD.toFixed(2)}`,
+          );
+
+          // Get org admins and owners for email notifications
+          const adminMemberships = await prisma.organizationMembership.findMany(
+            {
+              where: {
+                orgId: orgId,
+                role: { in: [Role.OWNER, Role.ADMIN] },
+              },
+              include: {
+                user: {
+                  select: { email: true },
+                },
+              },
+            },
+          );
+
+          const adminEmails = adminMemberships
+            .map((m) => m.user?.email)
+            .filter((email): email is string => Boolean(email));
+
+          const detectedAt = new Date();
+          const detectedAtUtc = detectedAt.toISOString().replace(".000Z", "Z");
+          if (adminEmails.length > 0) {
+            try {
+              // Send email notifications
+              await sendCloudSpendAlertEmail({
+                env,
+                orgId: orgId,
+                orgName: org.name,
+                alertTitle: alert.title,
+                currentSpend: currentSpendUSD,
+                threshold: thresholdUSD,
+                // casting due to cross-package type lag; property is supported in implementation
+                detectedAtUtc,
+                recipients: adminEmails,
+              } as any);
+
+              recordIncrement(
+                "langfuse.queue.cloud_spend_alert_queue.emails_sent",
+                1,
+                { unit: "emails" },
+              );
+
+              logger.info(
+                `[CLOUD SPEND ALERTS] Sent alert emails to ${adminEmails.length} recipients for org ${orgId}`,
+              );
+            } catch (e) {
+              recordIncrement(
+                "langfuse.queue.cloud_spend_alert_queue.email_failures",
+                1,
+                { unit: "emails" },
+              );
+              throw e;
+            }
+          }
+
+          // Update triggeredAt timestamp (reuse detection time)
+          await prisma.cloudSpendAlert.update({
+            where: { id: alert.id },
+            data: { triggeredAt: detectedAt },
+          });
+
+          countTriggeredAlerts++;
+        } else {
+          logger.debug(
+            `[CLOUD SPEND ALERTS] Alert ${alert.id} for org ${orgId} already triggered this billing cycle`,
+          );
+        }
+      } else {
+        // Reset triggeredAt if we're in a new billing cycle and threshold is not breached
+        if (alert.triggeredAt && alert.triggeredAt < currentPeriodStart) {
+          await prisma.cloudSpendAlert.update({
+            where: { id: alert.id },
+            data: { triggeredAt: null },
+          });
+          logger.debug(
+            `[CLOUD SPEND ALERTS] Reset alert ${alert.id} for org ${orgId} - new billing cycle`,
+          );
+        }
```

Comment:
It's odd to have this within the same if/else block. IMO we should move this to someplace else in the execution chain or make this a separate statement block. At least it's "surprising" that this behaviour comes here.

Reply:
oke